autoload/go/test.vim

```diff
@@ -122,13 +122,53 @@ function! go#test#Func(bang, ...) abort
   call call('go#test#Test', args)
 endfunction
 
+" Testcase runs a single subtest of a function for a subtest name on the
+" current cursor line.
+" Arguments are passed to the `go test` command.
+function! go#test#Case(bang, ...) abort
+  let l:testParent = go#util#TestName()
+  if l:testParent == ""
+    call go#util#EchoWarning("[test] no test found immediate to cursor")
+    return
+  endif
+
+  " Check for a struct initialization on the line, and take its value
+  " e.g. `name: \"test name\",` would extract 'testName'
+  let l:subTestName = go#test#subtestUnderCursor(getline("."))
+  if l:subTestName == ""
+    call go#util#EchoError('[test] no name for test case found on cursor line')
+    return
+  endif
+
+  " the pattern after l:testParent will allow testcases that are nested
+  " arbitrarily deep to be selected to run
+  let l:testParentPattern = l:testParent . "/\([^/]\+/\)*"
```

Comment:
No need to escape the `()` and `+` here. `go test` will handle the expression according to the rules at https://github.com/google/re2/wiki/Syntax. 

My apologies for leading you astray on that regexp syntax earlier. I always have to look it up to remember what I do and don't have to escape when using Go's regular expressions, because I spend way more time using Vim's regular expression engine than Go's.

Reply:
No worries! I'd rather iterate here than have someone (me) get hit with a bug that at best messes up a workflow or at worst hides failing tests.